src/arguments/CoreMessage.ts

```diff
@@ -1,17 +1,10 @@
-import {
-	GuildBasedChannelTypes,
-	isNewsChannel,
-	isTextChannel,
-	MessageLinkRegex,
-	SnowflakeRegex,
-	TextBasedChannelTypes
-} from '@sapphire/discord.js-utilities';
 import type { PieceContext } from '@sapphire/pieces';
-import { DMChannel, Message, NewsChannel, Permissions, Snowflake, TextChannel } from 'discord.js';
+import type { DMChannel, Message, NewsChannel, TextChannel, ThreadChannel } from 'discord.js';
+import { resolveMessage } from '../lib/resolvers';
 import { Argument, ArgumentContext, AsyncArgumentResult } from '../lib/structures/Argument';
 
 export interface MessageArgumentContext extends ArgumentContext {
-	channel?: DMChannel | NewsChannel | TextChannel;
+	channel?: DMChannel | NewsChannel | TextChannel | ThreadChannel;
 }
```

Comment:
Make sure this interface is exported at top level

Reply:
Shouldn't this be `Message['channel']` btw?

Reply:
I changed it to `Message['channel']` so it simplified it a lot, and I putted it inline (directly in the method's signature). I hope that's okay